src/files/ThemeXml.ts

```diff
@@ -0,0 +1,149 @@
+						}
+					}
+				}
+			}</a:theme>`,
+			{
+				fontSchemeName: this.themeElements.fontScheme.name,
+				majorFontLatinTypeface: this.themeElements.fontScheme.majorFont.latinFont.typeface,
+				majorFontLatinPanose: this.themeElements.fontScheme.majorFont.latinFont.panose,
+				majorOtherFonts: this.themeElements.fontScheme.majorFont.otherFonts,
+				minorFontLatinTypeface: this.themeElements.fontScheme.minorFont.latinFont.typeface,
+				minorFontLatinPanose: this.themeElements.fontScheme.minorFont.latinFont.panose,
+				minorOtherFonts: this.themeElements.fontScheme.minorFont.otherFonts
+			},
+			true
+		);
+	}
+
+	/**
+	 * Instantiate this class by looking at the DOCX XML for it.
+	 */
+	public static async fromArchive(archive: Archive, location?: string, xml?: Document): Promise<ThemeXml> {
+		// If a location is supplied, use that, otherwise use the default location for theme files.
+		location = location ?? 'word/theme/theme1.xml';
+		const themeDocument = archive.hasFile(location) ? await archive.readXml(location) : xml;
+		const fontScheme: Record<string, string | Font[]> = evaluateXPathToMap(`
+			//${QNS.a}theme/${QNS.a}themeElements/${QNS.a}fontScheme/map {
+				"name": @name/string(),
+				"majorFontLatinTypeface": ${QNS.a}majorFont/${QNS.a}latin/@typeface/string(),
+				"majorFontLatinPanose": ${QNS.a}majorFont/${QNS.a}latin/@panose/string(),
+				"majorFontOthers": array{${QNS.a}majorFont/${QNS.a}font/map { "script": @script/string(), "typeface": @typeface/string()}},
+				"minorFontLatinTypeface": ${QNS.a}minorFont/${QNS.a}latin/@typeface/string(),
+				"minorFontLatinPanose": ${QNS.a}minorFont/${QNS.a}latin/@panose/string(),
+				"minorFontOthers": array{${QNS.a}minorFont/${QNS.a}font/map { "script": @script/string(), "typeface": @typeface/string()}}
+			}
+		`, themeDocument);
+
+		const majorFontOthersCollection: Font[] = typeof fontScheme["majorFontOthers"] !== 'string'
+			? fontScheme["majorFontOthers"].map((font: Font) => {
+				return { script: font["script"], typeface: font["typeface"] } as Font
+			}) : [{ script: '', typeface: '' } as Font];
+
+
+		const minorFontOthersCollection: Font[] = typeof fontScheme["minorFontOthers"] !== 'string'
+			? fontScheme["minorFontOthers"].map((font: Font) => {
+				return { script: font["script"], typeface: font["typeface"] } as Font
+			}) : [{ script: '', typeface: '' } as Font];
+
+		const newFontScheme: FontScheme = {
+			name: fontScheme["name"] as string,
+			majorFont: {
+				latinFont: {
+					typeface: fontScheme["majorFontLatinTypeface"],
+					panose: fontScheme["majorFontLatinPanose"]
+				} as LatinFont,
+				otherFonts: majorFontOthersCollection
+			},
+			minorFont: {
+				latinFont: {
+					typeface: fontScheme["minorFontLatinTypeface"],
+					panose: fontScheme["minorFontLatinPanose"]
+				} as LatinFont,
+				otherFonts: minorFontOthersCollection
+			}
+		};
+
+		const newTheme = new ThemeXml(location);
+		newTheme.themeElements.fontScheme = newFontScheme;
```

Comment:
Ideally there is a setter for this object, ie. one that can (validate input and) allow users of `docxml` to construct a `ThemeXml` "manually" instead of only importing it from an existing Word file.

Since `FontScheme` is a sorta big object, it would be "user-friendlier" to have several setters; eg. `ThemeXml#setMajorFonts(latin: LatinFont, other: Font[])`, `ThemeXml#setMinorFonts(…)`.